cmake/ConfigureCompiler.cmake

```diff
@@ -346,9 +346,19 @@ else()
     set(DTRACE_PROBES 1)
   endif()
 
-  if(CMAKE_COMPILER_IS_GNUCXX)
-    set(USE_LTO OFF CACHE BOOL "Do link time optimization")
-    if (USE_LTO)
+  set(USE_LTO OFF CACHE BOOL "Do link time optimization")
+  if (USE_LTO)
+    if (CLANG)
+      if (LLVM_ENABLE_LTO STREQUAL "Full")
```

Comment:
Can this also be a cached variable?

Reply:
Oh, that's a good point.

By the way, @sfc-gh-zyang , `LLVM_ENABLE_LTO` is a macro LLVM understands.
https://clang.llvm.org/docs/ThinLTO.html#clang-bootstrap
Does that mean, we can set the cached variable `LLVM_ENABLE_LTO` as `Thin` by default, then we don't even need to add `add_compile_options` lines below, no?

Reply:
we have to add add_compile_options lines below, otherwise it doesn't work. According to https://clang.llvm.org/docs/ThinLTO.html#clang-bootstrap, we don't need the add_compile_options, I guess we need a newer cmake.

For LLVM_ENABLE_LTO, it's already defined in somewhere else, is it necessary to define it again?

Reply:
add CLANG_LTO_STRATEGY, please take another look.